src/main/java/seedu/address/model/person/Transaction.java

```diff
@@ -12,14 +12,14 @@
 public class Transaction {
 
     private final String description;
-    private final int amount;
+    private final double amount;
```

Comment:
If we are only accepting double of up to two decimal places, maybe need to have a method isValidAmount to check if the amount is valid

Reply:
There's a problem with forcing it to 2dp here. When we parse the user input and cast it to a double, because java uses IEE754 floating point conversion, it won't preserve the precision. This may cause transaction to reject valid transactions. Not sure how we should work around this. I think could either handle it in the parser, or we accept that there may be a little inaccuracy when doing arithmetic operations like `summary`

Reply:
I think we should ensure the user input for the amount is only up to 2 d.p. by having a isValidAmount method in the Transaction class, so the parser can call this method to check if the amount is valid. Can refer to other classes like Phone to see how they check if it's valid by using regex.

Reply:
Ok, got it. I misunderstood, I thought you meant to check within Transaction